lib/elixir/lib/stream.ex

```diff
@@ -1497,7 +1497,7 @@ defimpl Enumerable, for: Stream do
   end
 
   defp do_reduce(%Stream{enum: enum, funs: funs, accs: accs, done: done}, acc, fun) do
-    composed = :lists.foldl(fn fun, acc -> fun.(acc) end, fun, funs)
+    composed = :lists.foldl(fn f1, acc -> f1.(acc) end, fun, funs)
```

Comment:
I believe not shadowing is indeed better but `f1` is not a good name. Maybe `entry_fun`?